src/generic/Matrix.jl

```diff
@@ -369,6 +369,17 @@ canonical_unit(a::MatrixElem) = canonical_unit(a[1, 1])
 #
 ###############################################################################
 
+# linear indexing for row- or column- vectors
+Base.@propagate_inbounds function getindex(M::MatElem, x::Integer)
```

Comment:
```suggestion
Base.@propagate_inbounds function getindex(M::MatElem, x::Int)
```
I think we only support indexing with `Int`.

Reply:
For lower level indexing of AA/Nemo matrices, yes, but if that evolves, this implementation here is more future-proof, i.e. this functions just passes down the index and doesn't require specifically an `Int`. This also allows custom matrices to use other types of integers. But feel free to commit this bit